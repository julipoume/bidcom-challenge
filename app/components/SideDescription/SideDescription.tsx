"use client";

import React from "react";
import { DescriptionInfo } from "@/app/lib/definitions";

interface SideDescriptionProps {
  data: DescriptionInfo;
}

export default function SideDescription({ data }: SideDescriptionProps) {
  const { headline, title, description } = data;

  return (
    <aside className="flex flex-col items-center sm:items-start">
      {headline && (
        <h2 className="font-inter font-normal text-[12px] leading-[18px] sm:text-[14px] sm:leading-[20px] text-black px-[2px] py-[2px] tracking-[3px]">
          {headline}
        </h2>
      )}
      {title && (
        <h1
          className="font-inter font-normal text-[30px] leading-[36px] text-center tracking-normal sm:text-[48px] sm:leading-[48px] sm:text-left sm:tracking-[0%] text-black mb-4"
          dangerouslySetInnerHTML={{ __html: title }}
        />
      )}
      {description && (
        <p className="font-light text-[16px] leading-[24px] text-[#150142] mb-0 sm:mb-6 max-w-[547px] text-center sm:text-left">
          {description}
        </p>
      )}
    </aside>
  );
}
