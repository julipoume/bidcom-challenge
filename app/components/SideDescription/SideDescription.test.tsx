import React from "react";
import { render, screen } from "@testing-library/react";
import SideDescription from "./SideDescription";
import type { DescriptionInfo } from "@/app/lib/definitions";

describe("SideDescription", () => {
  const mockData: DescriptionInfo = {
    headline: "WELCOME",
    title: "<strong>Hello</strong> World",
    description: "This is a description text.",
  };

  it("snapshot", () => {
    const { container } = render(<SideDescription data={mockData} />);
    expect(container).toMatchSnapshot();
  });

  it("renders headline", () => {
    render(<SideDescription data={mockData} />);
    expect(screen.getByText("WELCOME")).toBeInTheDocument();
  });

  it("renders description", () => {
    render(<SideDescription data={mockData} />);
    expect(screen.getByText("This is a description text.")).toBeInTheDocument();
  });

  it("does not render elements if missing in data", () => {
    render(
      <SideDescription data={{ headline: "", title: "", description: "" }} />
    );
    expect(screen.queryByRole("heading", { level: 1 })).toBeNull();
    expect(screen.queryByText(/.+/)).toBeNull();
  });
});
