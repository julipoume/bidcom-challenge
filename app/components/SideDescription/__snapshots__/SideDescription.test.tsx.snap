// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`SideDescription snapshot 1`] = `
<div>
  <aside
    class="flex flex-col items-center sm:items-start"
  >
    <h2
      class="font-inter font-normal text-[12px] leading-[18px] sm:text-[14px] sm:leading-[20px] text-black px-[2px] py-[2px] tracking-[3px]"
    >
      WELCOME
    </h2>
    <h1
      class=" font-inter font-normal  text-[30px] leading-[36px] text-center tracking-normal sm:text-[48px] sm:leading-[48px] sm:text-left sm:tracking-[0%] text-black mb-4 "
    >
      <strong>
        Hello
      </strong>
       World
    </h1>
    <p
      class="font-light text-[16px] leading-[24px] text-[#150142] mb-0 sm:mb-6 max-w-[547px] text-center sm:text-left"
    >
      This is a description text.
    </p>
  </aside>
</div>
`;
